reactioncommerce/reaction
# Fix for "Receive error in Orders dashboard when using dollar discount codes

Resolves #4079  
Impact: **major**  
Type: **bugfix**

## Issue
The Order dashboard is trying to treat Discounts like other payment methods (which technically it is) and call `refund/list` on it, but of course that doesn't exist. Get this error on the client side console. 

```
{message: "Method 'code/refund/list' not found [404]", name: "Error", stack: "Error: Method 'code/refund/list' not found [404]↵ …7063aa833005aff7419b3f8c222e877582f1a3e0:2044:15)", code: undefined, signal: undefined}
hostname
```

## Solution / Changes

- Ensure that shopId is specified for the billing record when payment
method is "discount"
- Implement refund/list for payment provider, which essentially returns
an empty list (no refunds possible)

## Breaking changes
none expected

## Testing
1. Login as admin
2. Go the Payments/Codes dashboard
3. Add a new discount code with the discount method "credit" and the code to be "DOLLAROFF" and the amount to 1.
4. Add to cart and apply this discount when checking out
5. Go to the order dashboard and select that order
6. No error should be displayed in console. 
7. Capture payment
8. No `$NaN.undefined` should be displayed in invoice field `Adjusted Total`
